src/wled-device.ts

```diff
@@ -34,17 +59,101 @@ export class WledDevice extends Device {
+    );
+    this.addProperty(this.effectProperty);
+    this.paletteProperty = new PaletteProperty(
+      this, url, wledDescription.palettes
+    );
+    this.addProperty(this.paletteProperty);
+    this.effectSpeedProperty = new EffectSpeedProperty(this, url);
+    this.addProperty(this.effectSpeedProperty);
+    this.effectIntensityProperty = new EffectIntensityProperty(this, url);
+    this.addProperty(this.effectIntensityProperty);
+    // eslint-disable-next-line no-undefined
+    this.connected = undefined;
+    this.intervalMs = 1000;
   }
 
   public startPolling(intervalMs: number): void {
-    setInterval(() => this.poll(), intervalMs);
+    this.connectedNotify(true);
+    this.intervalMs = intervalMs;
+    setTimeout(() => this.poll(), intervalMs);
+  }
+
+  public connectedNotify(state: boolean): void {
+    super.connectedNotify(state);
+    this.connected = state;
+  }
+
+  public enterLiveMode(): void {
+    this.brightnessProperty.setReadOnly(true);
+    this.colorProperty.setReadOnly(true);
+    this.timerProperty.setReadOnly(true);
+    this.syncProperty.setReadOnly(true);
+    this.effectProperty.setReadOnly(true);
+    this.paletteProperty.setReadOnly(true);
+    this.effectSpeedProperty.setReadOnly(true);
+    this.effectIntensityProperty.setReadOnly(true);
+    this.getAdapter().handleDeviceAdded(this);
+  }
+
+  public leaveLiveMode(): void {
+    this.brightnessProperty.setReadOnly(false);
+    this.colorProperty.setReadOnly(false);
+    this.timerProperty.setReadOnly(false);
+    this.syncProperty.setReadOnly(false);
+    this.effectProperty.setReadOnly(false);
+    this.paletteProperty.setReadOnly(false);
+    this.effectSpeedProperty.setReadOnly(false);
+    this.effectIntensityProperty.setReadOnly(false);
+    this.getAdapter().handleDeviceAdded(this);
   }
 
   async poll(): Promise<void> {
-    const response = await fetch(this.url);
-    const json: WledDescription = await response.json();
+    let json: WledDescription;
+
+    try {
+      const response = await fetch(this.url);
+      json = await response.json();
+
+      if (json === null) {
+        setTimeout(() => this.poll(), this.intervalMs);
+        return;
+      }
+
+      if (this.connected === false) {
+        console.error('Reconnected');
+        this.connectedNotify(true);
+      }
+    } catch (ex) {
+      if (this.connected === true) {
+        console.error('Communication error: ', ex, '\nI will keep retrying!');
+        this.connectedNotify(false);
+      }
+
+      setTimeout(() => this.poll(), this.intervalMs);
+      return;
+    }
+
     this.onOffProperty.update(json);
     this.brightnessProperty.update(json);
     this.colorProperty.update(json);
+    this.liveProperty.update(json);
+    this.timerProperty.update(json);
+    this.syncProperty.update(json);
+    this.effectProperty.update(json);
+    this.paletteProperty.update(json);
+    this.effectSpeedProperty.update(json);
+    this.effectIntensityProperty.update(json);
```

Comment:
I think the control flow is easier to follow if there is only one setTimeout and one implicit return grimacing

```suggestion

      this.onOffProperty.update(json);
      this.brightnessProperty.update(json);
      this.colorProperty.update(json);
      this.liveProperty.update(json);
      this.timerProperty.update(json);
      this.syncProperty.update(json);
      this.effectProperty.update(json);
      this.paletteProperty.update(json);
      this.effectSpeedProperty.update(json);
      this.effectIntensityProperty.update(json);
    } catch (ex) {
      if (this.connected === true) {
        console.error('Communication error: ', ex, '\nI will keep retrying!');
        this.connectedNotify(false);
      }
    }
```

Reply:
Definitely!